src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.ServiceFabric.Client.Http;
+using ServiceFabricCommon = Microsoft.ServiceFabric.Common;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestHealthCheckPublisher : HealthCheckPublisher
+	{
+		private readonly RestClientWrapper m_clientWrapper;
+
+		private ServiceFabricHttpClient? m_client;
+
+		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
+
+		private string? m_nodeName;
+
+		internal static new string? HealthReportSourceId => nameof(RestHealthCheckPublisher);
+
+		internal const string NodeNameVariableName = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<ServiceFabricHealthCheckPublisher> logger,
+										IOptions<RestHealthCheckPublisherOptions> options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
+		{
+			m_nodeName = FindNodeName();
+			m_clientWrapper = new(new Uri(options.Value.RestHealthPublisherClusterEndpoint));
+			m_logger = logger;
+		}
+
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		{
+			if(m_nodeName == null)
+			{
+				m_logger.LogError(Tag.Create(), "Can't find node name.");
+				return;
+			}
+
+			m_client = await m_clientWrapper.GetAsync();
```

Comment:
You are already caching the client in a field once in the client wrapper, not need to save it here also, or alternatively remove the wrapper.

Reply:
afaik it is only reference, right? I am caching this secondly because it will be more efficient than calling it each time in `PublishHealthInfoAsync`

Reply:
Then what's the point of having `RestClientWrapper` class?

Reply:
Also is there any point in calling `m_clientWrapper.GetAsync()` each time instead of checking if it's already in `m_client` variable?

Reply:
RestClientWrapper is just abstraction used for more elegant solution

Reply:
Ad. https://github.com/microsoft/Omex/pull/334#discussion_r621062629
If we move null checking and recreating from wrapper to publisher, then there is no need for wrapper, but I think that this is more elegant but I'm open to change that if needed.

Reply:
As discussed offline this wrapper might be useful for other tasks and would be extracted into the Abstractions package in the follow-up.